### PR TITLE
[IMP] fleet: B2B review

### PR DESCRIPTION
**Pull request by Williambraecky**
Overall UI review of the fleet app.

Also changes the behaviour of shared fields between vehicles and vehicle
models.

Task ID: 2468228

### CODE REVIEW

File: addons/fleet/models/fleet_vehicle_model.py [hunk 1]
```diff
@@ -4,6 +4,19 @@
 from odoo import _, api, fields, models
 
 
+FUEL_TYPES = [
+        ('gasoline', 'Gasoline'),
```
Comment (kba-odoo) [inline addons/fleet/models/fleet_vehicle_model.py:8]: Could be de-indented

**Review by kbapt — CHANGES_REQUESTED** [review]

File: addons/fleet/models/fleet_vehicle_model.py (hunk 1 above)
Comment (Williambraecky) [reply addons/fleet/models/fleet_vehicle_model.py:8]: Yes sure no need to take that much space

File: addons/fleet/models/fleet_vehicle.py [hunk 2]
```diff
@@ -93,13 +94,24 @@ def _get_default_state(self):
...
     frame_type = fields.Selection([('diamant', 'Diamant'), ('trapez', 'Trapez'), ('wave', 'Wave')], help="Frame type of the bike")
-    electric_assistance = fields.Boolean()
+    electric_assistance = fields.Boolean(compute='_compute_model_fields', store=True, readonly=False)
     frame_size = fields.Float()
 
-    @api.depends('vehicle_type')
-    def _compute_doors(self):
-        for record in self:
-            record.doors = 5 if record.model_id.vehicle_type == 'car' else 0
+    @api.depends('model_id')
```
Comment (tivisse) [inline addons/fleet/models/fleet_vehicle.py:100]: Shouldn't we use instead:
```suggestion
    @api.depends(lambda self: ['model_id.%s' % (key) for key in MODEL_FIELDS_TO_VEHICLE.keys()])
```
?

Comment (Williambraecky) [reply addons/fleet/models/fleet_vehicle.py:100]: I believe it's better with just model_id, imagine you change the number of doors on `model_id` you would not want that change to be applied to existing vehicles, model_id acts more like a preset in this case

Comment (tivisse) [reply addons/fleet/models/fleet_vehicle.py:100]: Right ! :)

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**